### PR TITLE
Add logrus for structured logging.

### PR DESCRIPTION
**Pull request by tiegz**
This pulls in the `sirupsen/logrus` library that we're using in CLI, for a few reasons:

* so we can log to both stdout and stderr
* log-level based methods: Info(), Debug(), Fatal(), etc
* structured logging (explicitly with `log.WithFields`)

### Before

```
Starting Depper...
2021/01/14 11:39:52 Depper Publishing platform=rubygems name=blacklight-gallery version=3.0.0
2021/01/14 11:39:52 Depper Ingestor platform=npm sequence=11206853
2021/01/14 11:39:52 Depper Publishing platform=rubygems name=handcuffs version=1.4.1
2021/01/14 11:39:52 Depper Publishing platform=rubygems name=gtk_paradise version=0.4.302
2021/01/14 11:39:52 Depper Publishing platform=rubygems name=rusty_json_schema version=0.3.2
2021/01/14 11:40:01 Depper Publishing platform=rubygems name=asciidoctor-diagram-ditaamini version=0.13
2021/01/14 11:40:03 Depper Publishing platform=npm name=tidy-jsdoc-o2 version=0.0.7
2021/01/14 11:40:03 Sidekiq Publisher depper:ingest:npm:tidy-jsdoc-o2:0.0.7
...
^[[B^CCaught signal: interrupt.
```

### After

```
INFO[2021-01-14T11:36:11-05:00] Starting Depper                              
INFO[2021-01-14T11:36:11-05:00] Depper ingest                                 platform=rubygems
INFO[2021-01-14T11:36:13-05:00] Depper publish                                name=blacklight-gallery platform=rubygems version=3.0.0
INFO[2021-01-14T11:36:13-05:00] Depper ingest                                 platform=npm sequence=11206830
INFO[2021-01-14T11:36:13-05:00] Depper publish                                name=handcuffs platform=rubygems version=1.4.1
INFO[2021-01-14T11:36:14-05:00] Depper publish                                name=mayur-practice platform=npm version=1.4.1
INFO[2021-01-14T11:36:14-05:00] Depper publish                                name=serverless-resize-bucket-images platform=npm version=2.103.1
INFO[2021-01-14T11:36:14-05:00] Depper publish                                name=ewallet-reloader platform=npm version=2.103.1
...
^CINFO[2021-01-14T11:36:16-05:00] Exiting                                       signal=interrupt
```

https://app.clubhouse.io/tidelift/story/15442/depper-more-consistent-logging

### CODE REVIEW

File: ingestors/npm.go [hunk 1]
```diff
@@ -43,9 +44,9 @@ type NPMChangeDoc struct {
 func (ingestor *NPM) Ingest(results chan data.PackageVersion) {
 	since, err := ingestor.GetLatestSequence()
 	if err != nil {
-		log.Fatal(err)
+		log.WithFields(log.Fields{"ingestor": "npm"}).Fatal(err)
 	}
-	log.Printf("Depper Ingestor platform=npm sequence=%s", since)
+	log.WithFields(log.Fields{"platform": "npm", "sequence": since}).Info("Depper ingest")
```
Comment (tiegz) [inline ingestors/npm.go:49]: @jsonperl @mellisio what's your general take on doing this instead of printf-style interpolation?

Comment (jsonperl) [reply ingestors/npm.go:49]: Love it, what's the output look like in the log?

**Review by mellisio — APPROVED** [review]

File: ingestors/npm.go (hunk 1 above)
Comment (tiegz) [reply ingestors/npm.go:49]: `INFO[2021-01-14T11:36:13-05:00] Depper ingest                                 platform=npm sequence=11206830`

Not a huge fan of that big padding in the middle but it's not the end of the world.

File: main.go [hunk 2]
```diff
@@ -82,3 +78,15 @@ func (depper *Depper) registerIngestorStream(ingestor ingestors.StreamingIngesto
 		}
 	}()
 }
+
+func setupLogger() {
+	log.SetFormatter(&log.TextFormatter{
```
Comment (jsonperl) [inline main.go:83]: This seemed to control the logger globally yeah?

Comment (tiegz) [reply main.go:83]: Yeah, I'm still using the singleton logger here -- cool thing is that we can switch to the [Zalgo logger](https://github.com/sirupsen/logrus#formatters) if we want 😃

File: ingestors/npm.go (hunk 1 above)
Comment (jsonperl) [reply ingestors/npm.go:49]: Yea what's that padding all about?

Comment (tiegz) [reply ingestors/npm.go:49]: Can't figure out where that's set ATTM, but maybe we can revisit it later if it bugs us?

**Review by jsonperl — APPROVED** [review]

**Pull request merged**